### PR TITLE
feat(server): Pool lua interpreters

### LINKED ISSUE #741: Pool interpreters to reduce lock contention

### PR DESCRIPTION
**Pull request by dranikpg**
Fixes #741

### CODE REVIEW

File: src/server/common.cc [hunk 1]
```diff
@@ -25,6 +27,8 @@ extern "C" {
 #include "server/server_state.h"
 #include "server/transaction.h"
 
+ABSL_FLAG(uint32_t, interpreter_per_thread, 5, "Lua interpreters per thread");
+
```
Comment (dranikpg) [inline src/server/common.cc:31]: its strange common.cc is responsible for initializing server state 🤨

Comment (romange) [reply src/server/common.cc:31]: I agree. pull it out in another pr

Comment (dranikpg) [reply src/server/common.cc:31]: We can also increase it to about 10-15, above it usually delivers only small gains (given that most scripts are most likely a few ops), but I'm not sure why to keep the values low, it the interpreter overhead is likely super small

Comment (romange) [reply src/server/common.cc:31]: Lets make it 10

**Review by romange — APPROVED** [review]

**Review by romange — APPROVED** [review]

**Pull request merged**